File: 7월/2022-07-14/programmers_프린터_말감.py
from collections import deque

def solution(priorities, location):
    answer = 1
    queue = deque()
    for p in range(len(priorities)) :
        queue.append([priorities[p], p])
    temp = sorted(priorities, reverse=True)
    while True :
        q = queue.popleft()
        if q[0] != temp[0] :
            queue.append(q)
            continue
        if q[1] == location :
            break
        temp = temp[1:]
        answer += 1
    return answer

File: 8월/2022-09-04/programmers_프린터_말감.js
function solution(priorities, location) {
  let answer = 1;
  const document = priorities.map((el, index) => [el, index]);
  while (true) {
    const maxArr = document.slice(1).filter((el) => el[0] > document[0][0]);
    const temp = document.shift();
    if (maxArr.length) document.push(temp);
    else {
      if (temp[1] === location) break;
      answer += 1;
    }
  }
  return answer;
}
